lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -633,23 +635,27 @@ def create_configured_data_providers(lookup_invocation, parent_data_provider, us
       entry_datadir = @config_root + (he[KEY_DATADIR] || datadir)
       entry_datadir = Pathname(interpolate(entry_datadir.to_s, lookup_invocation, false))
       location_key = LOCATION_KEYS.find { |key| he.include?(key) }
-      locations = case location_key
-      when KEY_PATHS
-        resolve_paths(entry_datadir, he[location_key], lookup_invocation, @config_path.nil?)
-      when KEY_PATH
-        resolve_paths(entry_datadir, [he[location_key]], lookup_invocation, @config_path.nil?)
-      when KEY_GLOBS
-        expand_globs(entry_datadir, he[location_key], lookup_invocation)
-      when KEY_GLOB
-        expand_globs(entry_datadir, [he[location_key]], lookup_invocation)
-      when KEY_URIS
-        expand_uris(he[location_key], lookup_invocation)
-      when KEY_URI
-        expand_uris([he[location_key]], lookup_invocation)
-      when KEY_MAPPED_PATHS
-        expand_mapped_paths(entry_datadir, he[location_key], lookup_invocation)
-      else
-        nil
+      locations = nil
+      Puppet.override(avoid_hiera_lookup_errors: Puppet.lookup(:during_compilation){false}) do
```

Comment:
Is this because when we do a compile (the first time) we lazily load hiera configuration and we don't want to raise during these lookup invocations? I assume the second, etc compile don't call into this method since the `configured_data_provider(s)` have been loaded?

Reply:
I was never actually able to understand how it actually resolves these values for the manifest since the initial hiera interpolation fails in the `config5_interpolation` test; but yes, puppet initially tries to parse the hiera config and cannot interpolate values not yet set, so if puppet is compiling AND reading hiera config, then it will only warn once for variables not found. That's what I meant earlier when I said the logic was inverted.

It must be that the scope of the lookup eventually returns to the top level scope later on when it is parsing the manifest...